tests/tracer/test_sampler.py

```diff
@@ -261,6 +261,88 @@ def test_sampling_rule_init():
     assert rule.name == name_regex
 
 
+def test_sampling_rule_init_via_env():
+    # Testing single sampling rule
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+        assert len(sampling_rule) == 1
+
+    # Testing multiple sampling rules
+    with override_env(
+        dict(
+            DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}, \
+            {"sample_rate":0.5,"service":"my-service","name":"my-name"}]'
+        )
+    ):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+
+        assert sampling_rule[1].sample_rate == 0.5
+        assert sampling_rule[1].service == "my-service"
+        assert sampling_rule[1].name == "my-name"
+        assert len(sampling_rule) == 2
+
+    # Testing for only Sample rate being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+        assert len(sampling_rule) == 1
+
+    # Testing for no name being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+        assert len(sampling_rule) == 1
+
+    # Testing for no service being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == "abc"
+        assert len(sampling_rule) == 1
+
+    # Testing for Sample rate greater than 1.0
+    with pytest.raises(ValueError) as excinfo:
+        with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":2.0,"service":"xyz","name":"abc"}]')):
+            sampling_rule = DatadogSampler().rules
+    assert "SamplingRule(sample_rate={!r}) must be greater than or equal to 0.0 and less than or equal to 1.0" in str(
```

Comment:
Seems like `{!r}` is wrong ? should it say `2.0` ?

Reply:
Does this need to be `"" in str()` or can it be `"" == str(excinfo.value)` ?

Reply:
The {!r} was due to the sampling rule exception. I updated that line to include the actual sampling value
```
# Enforce sample rate constraints
        if not 0.0 <= sample_rate <= 1.0:
            raise ValueError(
                "SamplingRule(sample_rate={}) must be greater than or equal to 0.0 and less than or equal to 1.0".format(
                    sample_rate
                ),
            )
```

Reply:
I'll also add in the entire error message so we can use `"" == str(excinfo.value)`

Reply:
I ended up using the following workaround since comparing straight string values doesn't work in lower version of python due to the json not preserving load order.
```
assert "SamplingRule(sample_rate=2.0) must be greater than or equal to 0.0 and less than or equal to 1.0" in str(
        excinfo.value
    )
assert '"sample_rate": 2.0' in str(excinfo.value)
assert '"service": "xyz"' in str(excinfo.value)
assert '"name": "abc"' in str(excinfo.value)
```